CompilerSource/languages/language_adapter.h

```diff
@@ -35,12 +35,12 @@ struct language_adapter {
   virtual string get_name() = 0;
 
   // Sizable utilities
-  virtual int link_globals(parsed_object*, EnigmaStruct*, parsed_script*[], vector<parsed_script*>& tlines) = 0;
-  virtual int link_ambiguous(parsed_object*, EnigmaStruct*, parsed_script*[], vector<parsed_script*>& tlines) = 0;
+  virtual int link_globals(parsed_object*, EnigmaStruct*, vector<parsed_script*>&, vector<parsed_script*>& tlines) = 0;
+  virtual int link_ambiguous(parsed_object*, EnigmaStruct*, vector<parsed_script*>&, vector<parsed_script*>& tlines) = 0;
 
   // IDE_EDITABLEs added before compile
-  virtual int compile_parseAndLink(EnigmaStruct*,parsed_script*[], vector<parsed_script*>& tlines, const std::set<std::string>& script_names) = 0;
-  virtual int compile_parseSecondary(map<int,parsed_object*>&,parsed_script*[],int scrcount, vector<parsed_script*>& tlines, map<int,parsed_room*>&,parsed_object*, const std::set<std::string>&) = 0;
+  virtual int compile_parseAndLink(EnigmaStruct*, vector<parsed_script*>&, vector<parsed_script*>& tlines, const std::set<std::string>& script_names) = 0;
```

Comment:
Prefer `const&` so it's clear nothing is being mutated (the stuff behind the pointers may be, but that's fine).